synapse/module_api/__init__.py

```diff
@@ -1419,6 +1419,26 @@ async def set_push_rule_action(
             user_id, spec, {"actions": actions}
         )
 
+    async def get_monthly_active_users_by_service(
+        self, start_timestamp: Optional[int] = None, end_timestamp: Optional[int] = None
+    ) -> List[Tuple[str, str]]:
+        """Generates list of monthly active users and their services.
+        Please see corresponding storage docstring for more details.
+
+        Arguments:
+            start_timestamp: If specified, only include users that were first active
+                at or after this point
+            end_timestamp: If specified, only include users that were first active
+                at or before this point
+
+        Returns:
+            A list of tuples (appservice_id, user_id)
```

Comment:
Can you document what is returned for `appservice_id` if the user isn't from an appservice? (As per below I think we should return `None` in that case)

Reply:
I was hoping [this](https://github.com/buffless-matt/synapse/blob/451d7becda6c3e4971771bf12bce6aaba27a2315/synapse/storage/databases/main/monthly_active_users.py#L129-L130) sentence would alleviate the need to repeat what is mentioned in `get_monthly_active_count_by_service`, but for the sake of reducing the coupling between the two docstrings, I've added it here: 95accd8

Reply:
Ah, I see. I think the docstring should either document the return values fully, or not at all.